vcd/resource_vcd_nsxt_edgegateway_dhcp_forwarding.go

```diff
@@ -0,0 +1,222 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdNsxtEdgegatewayDhcpForwarding() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdNsxtEdgegatewayDhcpForwardingCreateUpdate,
+		UpdateContext: resourceVcdNsxtEdgegatewayDhcpForwardingCreateUpdate,
+		ReadContext:   resourceVcdNsxtEdgegatewayDhcpForwardingRead,
+		DeleteContext: resourceVcdNsxtEdgegatewayDhcpForwardingDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNsxtEdgegatewayDhcpForwardingImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"edge_gateway_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway ID for DHCP forwarding configuration",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Status of DHCP Forwarding for the Edge Gateway",
+			},
+			"dhcp_servers": {
+				Type:     schema.TypeSet,
+				Required: true,
+				// DHCP forwarding supports up to 8 IP addresses
+				MaxItems:    8,
+				Description: "IP addresses of the DHCP servers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceVcdNsxtEdgegatewayDhcpForwardingCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
I'm not sure how to do this without creating more functions like 
```go
func resourceVcdNsxtEdgegatewayDhcpForwardingCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}, operation string) diag.Diagnostics {
  <...>
}

func resourceVcdNsxtEdgegatewayDhcpForwardingCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
       return resourceVcdNsxtEdgegatewayDhcpForwardingCreateUpdate(ctx, d, meta, "create")
}

func resourceVcdNsxtEdgegatewayDhcpForwardingUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
       return resourceVcdNsxtEdgegatewayDhcpForwardingCreateUpdate(ctx, d, meta, "update")
}
```

As it seems a bit too much duplication, considering we use the same format in other resources, e.g `resourceVcdNsxtEdgegatewayRateLimiting`

Reply:
Personally, I prefer this approach (3 functions), which will give us better information when faced with troubleshooting.
Others may disagree. 
CC: @adambarreiro @Didainius

Reply:
Yeah, I agree that it helps when you need to troubleshoot an issue, otherwise you have to dig deeper to see in which exact operation the problem is.
Despite it looks a bit too much at first glance, I think it's helpful in the long term.

Reply:
Updated, split it into two functions as mentioned before.